File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/services/AuthService.kt
package com.desktopkitchen.pos.services

import com.desktopkitchen.pos.models.Employee
import com.desktopkitchen.pos.networking.api.AuthApi
import com.desktopkitchen.pos.networking.api.LoginRequest
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AuthService @Inject constructor(
    private val authApi: AuthApi
) {
    suspend fun login(pin: String): Employee {
        return authApi.login(LoginRequest(pin = pin))
    }
}

File: apps/pos/native-windows/DesktopKitchenPOS/Services/AuthService.cs
using DesktopKitchenPOS.Models;
using DesktopKitchenPOS.Networking.Endpoints;

namespace DesktopKitchenPOS.Services;

public class AuthService
{
    public async Task<Employee> LoginAsync(string pin)
    {
        return await DesktopKitchenPOS.App.Api.RequestAsync<Employee>(
            HttpMethod.Post, AuthEndpoints.Login, new LoginRequest { Pin = pin });
    }
}
